modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,264 @@
+    res2 = send_request_cgi({
+      'uri' => normalize_uri(datastore['PATH'],"#{url}#{match}~#{digit}.#{ext}*"),
+      'method' => @verb
+    })
+    return res2.code
+  end
+
+  def get_complete_status(url, digit , ext)
+    # Check if the file/folder name is less than 6 and complete
+    res2 = send_request_cgi({
+      'uri' => normalize_uri(datastore['PATH'],"#{url}*~#{digit}.#{ext}"),
+      'method' => @verb
+    })
+    return res2.code
+  end
+
+  def scanner
+    while !@queue_ext.empty?
+      f = @queue_ext.pop
+      url = f.split(':')[0]
+      ext = f.split(':')[1]
+      # Split string into name and extension and check status
+      status = get_incomplete_status(url, "*" , "1" , ext)
+      next unless status == 404
+      next unless ext.size <= 3
+
+      @charset_duplicates.each do |x|
+        if get_complete_status(url, x , ext) == 404
+          @files << "#{url}*~#{x}.#{ext}*"
+        end
+      end
+
+      if ext.size < 3
+        for c in @charset_extensions
+          @queue_ext << (f + c )
+        end
+      end
+    end
+  end
+
+  def scan
+    while !@queue.empty?
+      url = @queue.pop
+      status = get_status(url , "1" , "*")
+      # Check strings only upto 6 chars in length
+      next unless status == 404
+      if url.size == @name_size
+        @charset_duplicates.each do |x|
+          if get_status(url , x , "") == 404
+            @dirs << "#{url}*~#{x}"
+          end
+        end
+        # If a url exists then add to new queue for extension scan
+        for ext in @charset_extensions
+          @queue_ext << ( url + ':' + ext )
+          @threads << framework.threads.spawn("scanner", false) { scanner }
+        end
+      else
+        @charset_duplicates.each do |x|
+          if get_complete_status(url, x , "") == 404
+            @dirs << "#{url}*~#{x}"
+            break
+          end
+        end
+        if get_incomplete_status(url, "" , "1" , "") == 404
+          for ext in @charset_extensions
+            @queue_ext << ( url + ':' + ext )
+            @threads << framework.threads.spawn("scanner", false) { scanner }
+          end
+        elsif url.size < @name_size
+          for c in @charset_names
+            @queue  <<(url +c)
+          end
+        end
+      end
+    end
+  end
+
+  def reduce
+    # Reduce the total charset for filenames by checking if a character exists in any of the files
+    for c in @alpha.chars
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*#{c}*~1*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @charset_names << c
+      end
+    end
+  end
+
+  def ext
+    # Reduce the total charset for extensions by checking if a character exists in any of the extensions
+    for c in @alpha.chars
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*~1.*#{c}*"),
+        'method' => @verb
+      })
+      if res.code == 404
```

Comment:
`send_request_cgi` can return `nil`.

There should be some error handing here. The lazy approach would be:

```suggestion
      if res && res.code == 404
```